File: rust/src/bin/14/14.longest-common-prefix.rs
/*
 * @lc app=leetcode id=14 lang=rust
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {

        let mut strs_vec = vec![];
        let mut min_len = std::usize::MAX;

        strs.iter().for_each(|el| {
            let a = el.chars();
            let vec_char = el.chars().collect::<Vec<char>>();
            strs_vec.push(vec_char.clone());
            if vec_char.len() < min_len {
                min_len = vec_char.len();
            }
        });

        let mut ans = vec![];

        for i in 0..min_len {
            let p = strs_vec[0][i];
            let mut is_same = true;

            for j in 0..strs_vec.len() {
                if p != strs_vec[j][i] {
                    is_same = false;
                    break;
                }
            }
            if is_same {
                ans.push(p);
            } else {
                break;
            }
        }

        ans.iter().collect::<String>()

    }
}
// @lc code=end

File: typescript/14/14.longest-common-prefix.ts
/*
 * @lc app=leetcode id=14 lang=typescript
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
function longestCommonPrefix(strs: string[]): string {

    if (strs.length === 0) return "";
    if (strs.length === 1) return strs[0];

    let ans = "";

    for (let i = 0; i < 200; i++) {
        let str = strs[0][i];
        if (typeof str === "undefined") {
            break;
        }
        let is_same = true;

        for (let index = 0; index < strs.length; index++) {
            if (strs[index][i] !== str) {
                is_same = false;
                break;
            }
        }

        if (is_same) {
            ans += str;
        } else {
            break
        }


    }

    return ans;

};
// @lc code=end
